Components/MineSharp.Protocol/Packets/Clientbound/Login/DisconnectPacket.cs

```diff
@@ -9,27 +11,40 @@ namespace MineSharp.Protocol.Packets.Clientbound.Login;
 ///     Disconnect packet for login
 ///     See https://wiki.vg/Protocol#Disconnect_.28login.29
 /// </summary>
-public class DisconnectPacket : IPacket
+/// <param name="Reason">The reason for being disconnected</param>
+public sealed record DisconnectPacket(Chat Reason) : IPacket
 {
     /// <inheritdoc />
-    public PacketType Type => PacketType.CB_Login_Disconnect;
-    
-    /// <summary>
-    ///     The reason for being disconnected
-    /// </summary>
-    public required Chat Reason { get; init; }
-    
+    public PacketType Type => StaticType;
+    /// <inheritdoc />
+    public static PacketType StaticType => PacketType.CB_Login_Disconnect;
+
     /// <inheritdoc />
     public void Write(PacketBuffer buffer, MinecraftData version)
     {
         // Disconnect (login) packet is always sent as JSON text component according to wiki.vg
         buffer.WriteString(Reason.ToJson().ToString());
     }
-    
+
+    public static readonly Identifier BrandIdentifier = Identifier.Parse("minecraft:brand");
+
     /// <inheritdoc />
     public static IPacket Read(PacketBuffer buffer, MinecraftData version)
     {
         var reason = buffer.ReadString();
-        return new DisconnectPacket() { Reason = Chat.Parse(reason) };
+        // Disconnect (login) packet is always sent as JSON text component according to wiki.vg
+        // But thats wrong.
+        // Sometimes there are to strings sent. The first is always "minecraft:brand" and the second the brand, typically "vanilla".
```

Comment:
i don't think this is behaviour of a vanilla minecraft server. two strings alone are never a valid chat component, and the vanilla client only handles disconnect packets with a single chat component (at least for JE 1.20.4 and before).

Reply:
I am running a vanilla 1.20.4 server now for testing (and directly connect to it) and i received that packet.
You can try it for yourself

Reply:
hm I am also running 1.20.4 vanilla and I haven't encountered this yet... Are you being disconnected after some time or right after you logged in?

Reply:
I receive this packet while logging in and I need to ignore it otherwise the bot would disconnect. After ignoring that packet everything works fine. I do not understand why the server is sending it but not terminating the stream afterwards. Maybe we need to decompile the vanilla server and check.

Reply:
Maybe it has something to do with that my test server is in offline mode. So I do not need to log in my bot.

Reply:
Any chance this could be a race condition when switching from GameState login to configuration?
For GameState.Login, the packet id of disconnect is 0
For GameState.Configuration, the packet id of PluginMessage is also 0

The vanilla server sends a PluginMessage with exactly the contents "minecraft:brand" and "vanilla"

Reply:
Oh. Thats it. Ok that explains why you don't get that weird disconnect packet. I will take a look at how to fix that. Thanks for figuring that out.

Reply:
I fixed the problem and removed the code that ignores the weird packet